### PR TITLE
Include layer opacity in Heatmap shader

### LINKED ISSUE #13145: Heatmap layer opacity option is ignored
**Describe the bug**
Setting the opacity option in `ol/layer/Heatmap` has no effect.

**To Reproduce**
Open the example https://codesandbox.io/s/simple-forked-novjn?file=/main.js used in #13124 where there seemed to be no problem with the heatmap overlying the OSM layer in the shared context.  Add `opacity: 0.1` to the layer options.  It has no effect.  It appears be be unrelated to the `preserveDrawingBuffer` override as removing that and refreshing has no effect.  Then change to the latest dev version - it still no no effect,  But go back to version 6.9.0 and it was working, and the opacity setting does work for a WebGL Points layer in the latest version.

**Expected behavior**
Visually as in  version 6.9.0.

### PR DESCRIPTION
**Pull request by mike-000**
Fixes #13145

### CODE REVIEW

**Comment by tschaub** [comment]
A couple of the rendering tests use the heatmap layer.  We should add a new regression test that covers heatmap layer opacity.

File: src/ol/layer/Heatmap.js [hunk 1]
```diff
@@ -285,19 +285,23 @@ class Heatmap extends BaseVector {
...
             uniform sampler2D u_gradientTexture;
+            uniform float u_opacity;
 
             varying vec2 v_texCoord;
 
             void main() {
               vec4 color = texture2D(u_image, v_texCoord);
-              gl_FragColor.a = color.a;
+              gl_FragColor.a = color.a * u_opacity;
               gl_FragColor.rgb = texture2D(u_gradientTexture, vec2(0.5, color.a)).rgb;
```
Comment (tschaub) [inline src/ol/layer/Heatmap.js:295]: Should we be sampling the gradient texture after applying the layer opacity or before?  Feels to me like it should be after, but I'm not looking at the results.

Comment (mike-000) [reply src/ol/layer/Heatmap.js:295]: You can test in https://codesandbox.io/s/simple-forked-yrj3i?file=/main.js  Doing it after makes 50% opacity look much less than I would expect.

Comment (jahow) [reply src/ol/layer/Heatmap.js:295]: Layer opacity should definitely be applied after the gradient texture sampling. Here the `color.a` value isn't meant to be an opacity but more of an intensity value, which can then be mapped to a color.

Comment (jahow) [reply src/ol/layer/Heatmap.js:295]: This makes sense as it is actually, the layer opacity is only applied to `gl_FragColor` and not `color.a` so it works well.

**Comment by jahow** [comment]
I realized that this documentation here is not up to date since it does not mention the `u_opacity` uniform:

https://github.com/openlayers/openlayers/blob/main/src/ol/renderer/webgl/PointsLayer.js#L101

Maybe you could use this PR to also update this doc? thanks @mike-000

**Review by jahow — APPROVED** [review]
Looking good @mike-000, thanks for the doc and added test!

**Comment by tschaub** [comment]
As an aside, we should change things so we never directly use user-provided names in the shaders.  Instead of having a list of reserved names, we can prefix any user-provided identifiers to avoid collisions (as we now do with the style "variables").

**Pull request merged**